net/tstun/wrap.go

```diff
@@ -619,7 +619,7 @@ type natFamilyConfig struct {
 	// dstAddrToPeerKeyMapper is the routing table used to map a given dst IP to
 	// the peer key responsible for that IP.
 	// It only contains peers that require a MasqueradeAsIP address.
-	dstAddrToPeerKeyMapper *table.RoutingTable
+	dstAddrToPeerKeyMapper *bart.Table[key.NodePublic]
```

Comment:
I think you can combine `dstMasqAddrs` and `dstAddrToPeerKeyMapper` and just have a single `*bart.Table[netip.Addr]`

Reply:
Done, thanks for your patch!